src/odemis/driver/test/tfsbc_test.py

```diff
@@ -48,17 +48,24 @@ class TestBeamShiftController(unittest.TestCase):
 
     @classmethod
     def setUpClass(cls):
-        cls.bc = tfsbc.BeamShiftController("DC Offset", None, PORT, "FT2OMDD5")
-        # Values found during testing on the hardware
-        md = ((-0.00027788219369730165, 0.0013604844623992785),
-              (0.00012699407486043473, -0.0006217507619527259),
-              (-0.0013604844623992785, -0.00027788219369730165),
-              (0.0006217507619527259, 0.00012699407486043473))
-        cls.bc.updateMetadata({model.MD_CALIB: md})
-
-    @classmethod
-    def tearDownClass(cls):
-        pass
+        scanner = simulated.GenericComponent(name="Multibeam XT Sim", role="e-beam",
+                                             vas={"beamShiftTransformationMatrix":
+                                                  # Values found during testing on the hardware
+                                                  {"value": ((-0.00027788219369730165, 0.0013604844623992785),
+                                                             (0.00012699407486043473, -0.0006217507619527259),
+                                                             (-0.0013604844623992785, -0.00027788219369730165),
+                                                             (0.0006217507619527259, 0.00012699407486043473)),
+                                                   "readonly": True,
+                                                   "unit": ""}})
+
+        dependencies = {"scanner": scanner}
+        cls.bc = tfsbc.BeamShiftController("DC Offset", None, PORT, "FT2OMDD5", dependencies)
+
+    def test_dependency_scanner(self):
+        """Test dependency on scanner."""
+        calib_md = self.bc.getMetadata()[model.MD_CALIB]
+        self.assertIsInstance(calib_md, list)
+        self.assertIsInstance(calib_md[0][0], float)
```

Comment:
Now that you specify the exact values of the matrix in the set-up I would specifically check: `self.assertEqual(calib_md[0][0], -0.00027788219369730165)`